src/filters/BallFilter.cpp

```diff
@@ -137,3 +161,52 @@ bool BallFilter::ballIsVisible() const {
     //If we extrapolated the ball for longer than 0.05 seconds we mark it not visible
     return (lastPredictTime - lastUpdateTime) < 0.05;
 }
+
+double BallFilter::calculateVelocity() const {
+    double xVel = kalman->state()[2];
+    double yVel = kalman->state()[3];
+    return sqrt(xVel * xVel + yVel * yVel);
+}
+
+void BallFilter::updateState()  {
+    //TODO: Tune values
+    const double thresholdRest = 0.05; // Velocity at which ball is considered resting
+    const double thresholdKick = 0.1; // Velocity at which ball is considered kicked
+    const double switchRatio = 0.67; // Phase transition point from slipping to rolling
+
+    double velocity = calculateVelocity();
+
+    switch(state) {
+        case ROLLING:
+            // When velocity is low, the ball rests
+            if (velocity < thresholdRest) {
+                state = RESTING;
+            }
```

Comment:
no break;?

Reply:
This is difficult to read: i would prefer if lines 188-191 were copied here  and then you break after the case.